### ISSUE TITLE
[bug] "Ignore Rule" actions don't consider JSX

### ISSUE THREAD

**Issue by tommy-mitchell** [xojs/vscode-linter-xo#145]
When using the `"Ignore Rule"` Quick Fix actions, the inserted comments don't consider whether or not they're in a JSX node:

<details>
<summary>Before</summary>
<p>

<img width="587" alt="image" src="https://github.com/user-attachments/assets/b141ace4-9e7e-476b-b65f-7592ea0a6dfd">

</p>
</details>

<details>
<summary>After</summary>
<p>

<img width="865" alt="image" src="https://github.com/user-attachments/assets/7e3709cd-e944-4fcf-9ae9-e43563ab24f8">

</p>
</details>

**Comment by spence-s** [comment]
I agree this would be super cool, but there isn't really a good way to tell if you are in a block of jsx or not in the extension. ESLint extension doesn't do this either. - I'll keep an eye out for a way to do this in the future, but I wouldn't want to sacrifice too much performance to check for a jsx node on every lint, if its even possible. 

Closing for now.
